src/builtin.c

```diff
@@ -1219,24 +1219,49 @@ static jv f_strptime(jq_state *jq, jv a, jv b) {
   memset(&tm, 0, sizeof(tm));
   const char *input = jv_string_value(a);
   const char *fmt = jv_string_value(b);
-  const char *end = strptime(input, fmt, &tm);
-
-  if (end == NULL || (*end != '\0' && !isspace(*end))) {
-    jv e = jv_invalid_with_msg(jv_string_fmt("date \"%s\" does not match format \"%s\"", input, fmt));
-    jv_free(a);
-    jv_free(b);
-    return e;
+  char *inputdup = strdup(input);
+  char *fmtdup = strdup(fmt);
+
+  int ms = -1;
+  char *pf = strstr(fmtdup, "%f");
+  if (pf) {
+    char *msfmtloc = pf - 1;
+    char delim = *msfmtloc;
+    *msfmtloc = '\0';
+    char *msloc = strrchr(inputdup, delim);
```

Comment:
The assumption here is that `delim` will not be used for anything else after the `%f`...  I'm not very comfortable with this.